### DISCUSSION TITLE
Setup Ziggy with Inertia and Vue

### DISCUSSION THREAD

**Discussion by jeroenvanrensen** [inertiajs/inertia#1018] [Help (Vue)] [answered]
Hello,

I'm using Laravel and I want to setup Ziggy with Inertia and Vue.

I installed the Ziggy composer package and added `@routes` to my layout file.

The Inertia documentation says I should add it to the Vue instance:

```js
Vue.prototype.$route = route
```

However, I don't understand exactly to put it.

This is my `app.js` file:

```js
import { createApp, h } from 'vue'
import { createInertiaApp, Link } from '@inertiajs/inertia-vue3'

createInertiaApp({
    resolve: name => require(`./Pages/${name}`),
    setup({ el, App, props, plugin }) {
        createApp({ render: () => h(App, props) })
            .use(plugin)
            .component('Link', Link)
            .mount(el)
    }
})
```

Hope anyone can help me out.

Thanks!

**Comment by ajnsn** [comment, marked as answer]
Hey @jeroenvanrensen - this may help: https://github.com/inertiajs/inertia/discussions/822

**Reply by jeroenvanrensen** [reply]
Hello, do you mean something like this?

```js
import { createApp, h } from 'vue'
import { createInertiaApp, Link } from '@inertiajs/inertia-vue3'

createInertiaApp({
    resolve: name => require(`./Pages/${name}`),
    setup({ el, App, props, plugin }) {
        createApp({ render: () => h(App, props) })
            .use(plugin)
            .use((Vue.prototype.$route = route)) // here I added it
            .component('Link', Link)
            .mount(el)
    }
})

```

**Reply by ajnsn** [reply]
Hey @jeroenvanrensen There are various ways to integrate a custom method.  When you want to `use()` a plugin, you might want to use the provided Ziggy Vue plugin ﻿https://github.com/tighten/ziggy#vue, e.g. `.use(ZiggyVue, Ziggy)` where `Ziggy` is your imported routes file.

In general this is nothing different with or without Inertia. You can add everything to the `createApp` chain as you would with Vue.

Note that the v2 `Vue.prototype` has changed in v3 to `app.config.globalProperties`, see here:
https://v3.vuejs.org/api/application-config.html#globalproperties

If you want to follow this "global" path, this might help as well:
https://github.com/inertiajs/inertia/discussions/818#discussioncomment-1032136

**Reply by jeroenvanrensen** [reply]
Thanks now it's working!
